tests/test_runtime_utils.py

```
@@ -65,6 +65,22 @@ def test_is_running_as_service_false_when_parent_is_not_systemd() -> None:
         assert runtime_utils.is_running_as_service() is False
 
 
+def test_is_running_as_service_false_when_ppid_is_zero() -> None:
+    """Invalid parent PID 0 should return False without reading /proc/0/comm."""
+    status_handle = mock_open(read_data="Name:\tpython\nPPid:\t0\n").return_value
+
+    def _side_effect(path: str, *args: Any, **kwargs: Any) -> IO[Any]:
```

Ruff (0.15.9)

[warning] Unused function argument: `args`

(ARG001)